packages/protocol/engine/tangle/booker/booker.go

```diff
@@ -229,31 +234,39 @@ func (b *Booker) block(id models.BlockID) (block *Block, exists bool) {
 	return storage.Get(id)
 }
 
-func (b *Booker) book(block *Block) (err error) {
+func (b *Booker) book(block *Block) (inheritingErr error) {
+	b.bookingMutex.Lock(block.ID())
```

Comment:
Add comment to explain that this is supposed to cover the synchronous execution of the "BlockBooked" event from the virtuavoting to add votes while mutually excluding the forking process.